tortoise.py

```diff
@@ -0,0 +1,9 @@
+def main():
+    with open('tiny.tortoise', 'r') as tiny_tortoise:
+        tortoise_program = tiny_tortoise.readlines()
+        for line in tortoise_program:
```

Comment:
Once the contents of the file have been read with `readlines()`, all the data from the file is in the computer's memory. There's no need to keep the file open after that point. The code which deals with the contents of `tortoise_program` should be indented left a level, to move it out of the `with` block.